### PR TITLE
move CI to go1.15 and go1.16

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
move CI to go1.15 and go1.16

## Motivation and Context
remove go1.14 CI builds

## How to test this PR?
everything should work as is

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Comment by minio-trusted** [comment]
```
Error running mint automation
+ mkdir -p 11570-03c71a6/gopath/src/github.com/minio
+ git -C 11570-03c71a6/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 11570-03c71a6/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 11570-03c71a6/gopath/src/github.com/minio/minio fetch --quiet minio pull/11570/head:pr11570
+ git -C 11570-03c71a6/gopath/src/github.com/minio/minio checkout --quiet pr11570
+ GO111MODULE=on
+ GOPATH=/home/mint-auto/mint-auto/11570-03c71a6/gopath
+ make -C 11570-03c71a6/gopath/src/github.com/minio/minio --quiet
can't load package: package .: build constraints exclude all Go files in /home/mint-auto/mint-auto/11570-03c71a6/gopath/src/github.com/minio/minio
make: *** [build] Error 1

```

File: main.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// +build go1.14
```
Comment (klauspost) [inline main.go:1]: Not needed. `go1.14` is set on all versions following v1.14.

Comment (klauspost) [reply main.go:1]: ...or is it used as a min version check? You can use the go.mod version for that.

Comment (harshavardhana) [reply main.go:1]: yes its a min version check, its old style removing it @klauspost

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#92e7fe6a-7b83-44ff-ac98-dd858e2e03d9) |
<a name="92e7fe6a-7b83-44ff-ac98-dd858e2e03d9"></a>
### 11570-de63e02/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c2.minio.io:30292
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 9ed3e941e701:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 5 seconds
(5/15) Running awscli tests ... FAILED in 33 seconds
{
  "name": "awscli",
  "duration": 2712,
  "function": "aws --endpoint-url http://minio-c2.minio.io:30292 s3api copy-object --bucket awscli-mint-test-bucket-26441 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-26441/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected 084e1383b70fb0c51acc680fef370023, got ac57de7156d7fc25ac1a65f81fa3989b"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 47 seconds
(7/15) Running minio-dotnet tests ... done in 52 seconds
(8/15) Running minio-go tests ... FAILED in 2 minutes and 32 seconds
{
  "args": {
    "destination": {
      "Bucket": "minio-go-test-9x0w43pu995pxdrw",
      "Object": "dstObject",
      "Encryption": {},
      "UserMetadata": null,
      "ReplaceMetadata": false,
      "UserTags": null,
      "ReplaceTags": false,
      "LegalHold": "",
      "Mode": "",
      "RetainUntilDate": "0001-01-01T00:00:00Z",
      "Size": 0,
      "Progress": null
    },
    "source": {
      "Bucket": "minio-go-test-9x0w43pu995pxdrw",
      "Object": "srcObject",
      "VersionID": "",
      "MatchETag": "",
      "NoMatchETag": "",
      "MatchModifiedSince": "0001-01-01T00:00:00Z",
      "MatchUnmodifiedSince": "0001-01-01T00:00:00Z",
      "MatchRange": false,
      "Start": 0,
      "End": 0,
      "Encryption": null
    }
  },
  "duration": 5019,
  "error": "We encountered an internal error, please try again.: cause(s2: corrupt input)",
  "function": "CopyObject(destination, source)",
  "message": "GetObject failed",
  "name": "minio-go: testUnencryptedToSSES3CopyObject",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 1 minutes and 41 seconds
{
  "name": "minio-java",
  "function": "copyObject()",
  "args": "[match etag]",
  "duration": 296,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = PreconditionFailed, message = At least one of the pre-conditions you specified did not hold, bucketName = minio-java-test-1i22r7p, objectName = minio-java-test-39cnsei-copy, resource = /minio-java-test-1i22r7p/minio-java-test-39cnsei-copy, requestId = 1664CD1658AC190C, hostId = 3f700732-890b-470f-8412-aa066d4101c2)\nrequest={method=PUT, url=http://minio-c2.minio.io:30292/minio-java-test-1i22r7p/minio-java-test-39cnsei-copy, headers=x-amz-copy-source-if-match: 71cff0a060f852067e443ad1e24ae26c-1\nx-amz-copy-source: /minio-java-test-ll1316/minio-java-test-39cnsei\nHost: minio-c2.minio.io:30292\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210218T092102Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210218/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=412, headers=Accept-Ranges: bytes\nContent-Length: 418\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nETag: \"71cff0a060f852067e443ad1e24ae26c\"\nLast-Modified: Thu, 18 Feb 2021 09:21:02 GMT\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 1664CD1658AC190C\nX-Xss-Protection: 1; mode=block\nDate: Thu, 18 Feb 2021 09:21:02 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.execute(MinioClient.java:563), io.minio.MinioClient.executePut(MinioClient.java:904), io.minio.MinioClient.copyObject(MinioClient.java:1232), FunctionalTest.testCopyObjectMatchETag(FunctionalTest.java:1850), FunctionalTest.copyObject(FunctionalTest.java:2016), FunctionalTest.runObjectTests(FunctionalTest.java:3757), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(8/15) Running minio-js tests ... done in 52 seconds
(9/15) Running minio-py tests ... done in 3 minutes and 18 seconds
(10/15) Running s3cmd tests ... FAILED in 6 seconds
{
  "name": "s3cmd",
  "duration": "3193",
  "function": "test_put_object_multipart",
  "status": "FAIL",
  "error": "WARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Too many failures. Giving up on '/mint/data/datafile-65-MB'\nERROR: \nUpload of '/mint/data/datafile-65-MB' part 1 failed. Use\n  /usr/local/bin/s3cmd abortmp s3://s3cmd-test-bucket-12131/s3cmd-test-object-17431 2068dc1b-b942-40f9-92e6-6db0baad7636\nto abort the upload, or\n  /usr/local/bin/s3cmd --upload-id 2068dc1b-b942-40f9-92e6-6db0baad7636 put ...\nto continue the upload.\nERROR: Upload of '/mint/data/datafile-65-MB' failed too many times (Last reason: )"
}
(10/15) Running s3select tests ... done in 8 seconds
(11/15) Running security tests ... done in 0 seconds

Executed 11 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Review by klauspost — APPROVED** [review]
LGTM

**Pull request merged**